views/eng-chinese-translator.ejs

```diff
@@ -238,48 +271,40 @@
             lambda.invoke(params, function (error, data) {
                 document.getElementById('spinner').style.display = 'none';
                 if (error) {
-                    console.error('AWS error: ', JSON.stringify(error));
+                    showError('AWS error: ' + JSON.stringify(error));
                     return;
                 }
                 if (data.FunctionError) {
-                    console.error("Lambda function error: ", data.Payload);
+                    showError("Lambda function error: " + data.Payload);
                     return;
                 }
                 if (data.Payload) {
-                    var response = JSON.parse(data.Payload);
+                    let response = JSON.parse(data.Payload);
                     if (response.statusCode === 200) {
-                        var translationData = response.body;
-                        document.getElementById('response').value = translationData.output;
-
-                        const detectedResponseLanguage = detectLanguage(translationData.output);
-                        document.getElementById("responseLabel").textContent = detectedResponseLanguage;
+                        document.getElementById('response').value = response.body.output;
+                        document.getElementById("userInputLabel").textContent = response.body.input_language;
+                        document.getElementById("responseLabel").textContent = response.body.output_language;
                     } else {
-                        console.log("lambda error: ", response.body)
+                        showError("lambda error: " + response.body)
                     }
                 }
             });
         }
 
-        // Detect the language of the user input and call the Lambda function
-        function callLambdaFunction(inputText) {
-            const detectedInputLanguage = detectLanguage(inputText);
-            document.getElementById("userInputLabel").textContent = detectedInputLanguage;
-
-            invokeLambdaFunction(inputText);
-        }
-
-        // Submit the user input to the Lambda function
         function handleSubmit() {
-            // Show the spinner when the 'Submit' button is clicked
             document.getElementById('spinner').style.display = 'inline-block';
-
-            callLambdaFunction(document.getElementById('userInput').value);
+            invokeLambdaFunction(document.getElementById('userInput').value);
         }
 
-        // Detect the language of the user input
```

Comment:
Since you deleted the `detectLanguage` function,  shouldn't you remove the calling of this function:
https://github.com/Carma-tech/CarmaTech-API/blob/main/views/eng-chinese-translator.ejs#L295~L296%20?